### PR TITLE
imported/added library specific PMD handling and fixed most dominant

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
* added pmd_rules.xml and corresponding project configs
* fixed variable naming (mostly 'private static final ...' fields)
* fixed resource leakage bugs identified by Coverity tool

### CODE REVIEW

File: chartfx-chart/src/main/java/de/gsi/chart/plugins/AbstractSingleValueIndicator.java [hunk 1]
```diff
@@ -23,12 +23,12 @@
      * The default distance between the data point coordinates and mouse cursor
      * that triggers shifting the line.
      */
-    public static final int DEFAULT_PICKING_DISTANCE = 30;
-    static final double MIDDLE_POSITION = 0.5;
-    static final String STYLE_CLASS_LABEL = "value-indicator-label";
-    static final String STYLE_CLASS_LINE = "value-indicator-line";
-    static final String STYLE_CLASS_MARKER = "value-indicator-marker";
-    protected static double TRIANGLE_HALF_WIDTH = 5.0;
+	protected static final int DEFAULT_PICKING_DISTANCE = 30;
```
Comment (wirew0rm) [inline chartfx-chart/src/main/java/de/gsi/chart/plugins/AbstractSingleValueIndicator.java]: Indentation

File: chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java [hunk 2]
```diff
@@ -24,192 +24,198 @@
...
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return null;
-    }
-
-    public static void main(final String[] args) {
-        Application.launch(args);
-    }
+	private static final int MAX_POINTS = 512;
```
Comment (wirew0rm) [inline chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java]: Indentation with tabs?

File: chartfx-samples/src/main/java/de/gsi/math/samples/WaveletScalogram.java [hunk 3]
```diff
@@ -183,33 +184,33 @@ public Node getContent() {
...
-            return ret;
-
-        } catch (Exception e) {
-            e.printStackTrace();
-            JOptionPane.showMessageDialog(null, "Could not data file:\n" + e.getMessage(), "Error",
-                    JOptionPane.ERROR_MESSAGE);
-        }
-        return null;
-    }
+	private double[] readDemoData(int index) {
```
Comment (wirew0rm) [inline chartfx-samples/src/main/java/de/gsi/math/samples/WaveletScalogram.java]: Indentation with tabs

File: chartfx-chart/src/main/java/de/gsi/chart/plugins/AbstractSingleValueIndicator.java (hunk 1 above)
Comment (RalphSteinhagen) [reply chartfx-chart/src/main/java/de/gsi/chart/plugins/AbstractSingleValueIndicator.java]: fixed (single line)

File: chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java (hunk 2 above)
Comment (RalphSteinhagen) [reply chartfx-samples/src/main/java/de/gsi/math/samples/WaveletDenoising.java]: fixed (file was from an old era where tabs were still acceptable ;-)8 )

File: chartfx-samples/src/main/java/de/gsi/math/samples/WaveletScalogram.java (hunk 3 above)
Comment (RalphSteinhagen) [reply chartfx-samples/src/main/java/de/gsi/math/samples/WaveletScalogram.java]: fixed (file was from an old era where tabs were still acceptable ;-)8 )

**Comment by RalphSteinhagen** [comment]
different pmd_rules.xml for each submodule because the PMD plugin does not allow for relative paths (ie. '../pmd_rules.xml') and because we can allow different styles for some of the modules (e.g. using System.out & System.err instead of logger in the sample module -> allows for lighter-weight testing and reading of code)... other changes are in

**Pull request merged**